src/robotlibcore.py

```diff
@@ -97,7 +98,27 @@ def run_keyword(self, name, args, kwargs=None):
         return self.keywords[name](*args, **(kwargs or {}))
 
     def get_keyword_arguments(self, name):
-        kw = self.keywords[name] if name != '__init__' else self.__init__
+        if robot_version >= '3.2':
+            return self.__new_arg_spec(name)
+        return self.__old_arg_spec(name)
+
+    def __new_arg_spec(self, keyword_name):
+        kw = self.__get_keyword(keyword_name)
+        if kw is None:
+            return [tuple(), ]
+        args, defaults, varargs, kwargs = self.__get_arg_spec(kw)
+        args = [(arg, ) for arg in args]
+        args += [(name, value) for name, value in defaults]
+        if varargs:
+            args.append(('*{}'.format(varargs), ))
+        if kwargs:
+            args.append(('**{}'.format(kwargs), ))
```

Comment:
There's no need to return other than defaults as tuples. That's ok but also `['mandatory, ('optional', 'default')]` is fine. That means only defaults need to be handled differently depending on RF versions and that means some duplication could be removed.

Looking at this code I remember that `'{}'.format(value)` fails on Python 2 with UnicodeErrorError if `value` is Unicode and contains non-ASCII stuff. With Python 2 varargs and kwards cannot contain such characters (they aren't even Unicode) but perhaps it would be good to go through the library code and change all `'{}'.format(value)`s to `u'{}'.format(value)` or `'%s' % value`.

Reply:
1) I thought all are tuples, will change.

2) I am already forgetting all the Python 2 quirks. Although that is almost same as the from the old format where `format()` has served pretty well. But it will be consistent with other code to use `%s`. Just can't wait to ditch Python 2.